app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -166,6 +166,23 @@ def update_cost
       affected_app: affected_app
     }, status: :ok and return
   end
+  def delete
+    @helm_infrastructure = HelmInfrastructure.find_by(cluster_name: params[:id])
+    
+    if @helm_infrastructure.blank? || !@helm_infrastructure.active?
+      render(json: {
+                 success: false,
+                 errors: ['Infrastructure not found'],
+                 code: 404,
+               }, status: :not_found) && return
+    end
+    
+    @helm_infrastructure.delete
```

Comment:
instead of hard deleting, I think you can call the content of this function https://github.com/BaritoLog/BaritoMarket/blob/master/app/controllers/helm_infrastructures_controller.rb#L77-L88